### ISSUE TITLE
Workfile tool to show all versions

### ISSUE THREAD

**Issue by m-u-r-p-h-y** [pypeclub/OpenPype#4216]
Workfile tool should be able to show all versions already registered in DB, regardless of DCC it is launched from.

Currently, it only shows workfiles with native extensions.

![image](https://user-images.githubusercontent.com/352795/207103976-0674e641-263d-4cb7-9cf0-ac25bf3b4910.png)

This situation leads to confusion when artists are saving workfiles from different DCCs under one task.

**Comment by BigRoy** [comment]
I think this could get ugly quickly - the `workspace.mel`, would that be a workfile too? Any other file that the user had saved in their work folder, or temporary files from the DCC - should they show too?

It's not super trivial to know what we'd consider "workfiles" which means that in the end the list would just become a file browser and list all files in the folder.

I'm also a bit worried that a user might get confused if there are listed files there that they can't interact with. Them being there, greyed out, starts giving of the assumption that there might be a way to unlock that.

I think the simplicity that's there now can already clutter quite a lot if there are many workfiles for the current DCC and if we can I'd refrain from showing more data that we can't interact with.

Would love to hear opinions from others! :)

**Comment by tokejepsen** [comment]
> This situation leads to confusion when artists are saving workfiles from different DCCs under one task.

I agre with @BigRoy that would lead to a file explorer replicate which would be confusing.

@m-u-r-p-h-y has this been reported by a client or community?

**Comment by m-u-r-p-h-y** [comment]
yes. You start working on a model in Blender and another artist will launch the workfile tool in Maya and he has no chance to know there was work done in other DCC,

**Comment by m-u-r-p-h-y** [comment]
workfile area is a bit messier than publish by design.  why do you find it confusing to see what the artist has in there? Should we hide the files? why?

**Comment by tokejepsen** [comment]
> You start working on a model in Blender and another artist will launch the workfile tool in Maya and he has no chance to know there was work done in other DCC,

That is actually a good use case and point!

I think we need a bit more of a think about the UX though, cause just having a bunch is files greyed out in between selectable files (thinking for example Maya and Nuke files) in the list will not be a very good user experience. Maybe we should be all the non-native workfiles in a list below the native workfiles?

![Capture](https://user-images.githubusercontent.com/1860085/224997315-77c35b67-3a25-460d-b0ed-878c3d0a747d.PNG)

**Comment by BigRoy** [comment]
Be aware that this would be extra complicated with something like the `AVALON_SCENEDIR` functionality which is something we're using. The files from our different DCCs actually rarely end up in the same work folder - as such, the functionality for us would just start showing `workspace.mel` as an extra file and nothing else useful.

> You start working on a model in Blender and another artist will launch the workfile tool in Maya and he has no chance to know there was work done in other DCC,

I agree that it'd be nice if we can somehow make that more apparent to the user, but I'm in doubt about what we're setting up as a solution now is the perfect solution.

If (non-published) workfiles get registered in the database then technically it'd be trivial to know about any workfiles, no matter the DCC for the current task. It could even highlight which DCC (and maybe even app name/version - created it?)

**Comment by tokejepsen** [comment]
> Be aware that this would be extra complicated with something like the AVALON_SCENEDIR functionality which is something we're using. The files from our different DCCs actually rarely end up in the same work folder (they could be in the same task, but not the same folder on disk) - as such, the functionality for us would just start showing workspace.mel as an extra file and nothing else useful.

Good point! Shame we cant query the `workio.py` > `file_extensions`, since they all import the host api. Technically `file_extensions` could be moved somewhere else I guess, but not sure whether that just complicates things.

**Comment by m-u-r-p-h-y** [comment]
we could have a button (next to published files toggle) with 3 modes/levels

1. show only files native to DCC I'm running the Workfile tool from 
2. show all DB registered versions (with native files on top as @tokejepsen suggested
3. show everything in workfile root

**Comment by movalex** [comment]
I think the Work Files dialogue should also show the DCC context somewhere above the file list, then it would not leave much place for confusion. But I agree, there should be a button to show other DCC's files grayed-out, with ability to read the other artist's note for those workfiles.

**Comment by tokejepsen** [comment]
> I think the Work Files dialogue should also show the DCC context somewhere above the file list, then it would not leave much place for confusion.

@movalex could you elaborate on this?

**Comment by movalex** [comment]
> could you elaborate on this?
@tokejepsen I think of adding a DCC information to the window title to remind where are you're currently at. Because at some point you can launch multiple WF dialogues from both DCC. Maybe adding a DCC icon somewhere would work too. 
Something like this:
![prototype](https://user-images.githubusercontent.com/11698866/226564592-ee6b118f-2937-480c-9c8b-56ec087d26b0.jpg)

> I think this could get ugly quickly - the workspace.mel, would that be a workfile too? Any other file that the user had saved in their work folder, or temporary files from the DCC - should they show too?

@BigRoy Only the files that are supported by other OP DCC should be visible with `Show other DCC workfiles` option enabled (I guess `Show All` checkbox would be shorter, but a bit confusing). The file list may also be limited by the task type. No need to show `.ma` files if the task type if `Compositing`. However `.aep` and `.nk` files should be visible just to get the idea of how the work is going.

> I'm also a bit worried that a user might get confused if there are listed files there that they can't interact with. Them being there, greyed out, starts giving of the assumption that there might be a way to unlock that.

To avoid that the `Show All` checkbox should be disabled by default

**Comment by BigRoy** [comment]
>  I think of adding a DCC information to the window title to remind where are you're currently at. Because at some point you can launch multiple WF dialogues from both DCC.

I think this is especially an issue with DCCs where the UI might not be parented to the application because the host isn't built with Qt, like e.g. Blender, Fusion or alike.

Having a little title mentioning what it's connected to, say "Fusion 18 > Project > Asset" could be nice.
We ended up on our own branch to have the OpenPype menu in Fusion be prefixed with the asset name so that if you were to ALT+TAB you'd instantly see the title about which window relates to what Fusion instance, etc.

So I can see the benefit here, but might be trickier to keep in sync over time (if we were to also show e.g. the asset)

> Only the files that are supported by other OP DCC should be visible with Show other DCC workfiles option enabled (I guess Show All checkbox would be shorter, but a bit confusing). The file list may also be limited by the task type. No need to show .ma files if the task type if Compositing. However .aep and .nk files should be visible just to get the idea of how the work is going.

Actually, I think this filtering per task is exactly what will again become confusing. The whole issue here is that the artist doesn't know about work done _elsewhere_ which could've been work outside of the scope of what they think the task is.

The same with maybe filtering by "workfile-compatible" files. It'll just increase complexity and maybe confuse the artist that a file is not shown at all? We'll need to keep into account applications that are enabled for the project?

> To avoid that the Show All checkbox should be disabled by default

If the files don't show by default would most artists even notice that files did exist elsewhere? Is anyone really going to explicitly check each time. The issue brought up here is that the artist didn't know anyone did any work on it yet.

I'm fine with it off by default - I just wonder whether it solves anything.

Also, that Show All checkbox would get disabled when going into "Show Published" mode I assume?

---

By the way, I think the greyed out items should at least provide a tooltip or something explaining that it's a file not compatible with the current application and that's why it's greyed out.

**Comment by movalex** [comment]
After all, I don't think the idea of showing all files of the current task folder is something bad. If that would be an option, there might be a better understanding on how the project is structured, and where the files are.  Of course, there's already a `Browse` button, that might show all files with native browse interface, but I personally never used that. It seems misleading, because we use Openpype to forget about the Explorer file management. 
Speaking of filtering, there might be a filter dropdown list, with 3 levels of filtering: `Show Current Workfiles (default)` -- `Show All DCC Files -- Show All Files`. If all files are shown, only the current DCC files would be active, others are greyed out. Right click on the greyed out files or folder might have a `Show in Finder/Explorer` option.  And the other checkbox could be `Filter Published`, that overrides the dropdown filter... My point is: why wouldn't give them users some options. Let them decide.
